api/v1alpha1/trafficextensionypolicy_types.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+const (
+	// KindTrafficExtensionPolicy is the name of the TrafficExtensionPolicy kind.
+	KindTrafficExtensionPolicy = "TrafficExtensionPolicy"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:shortName=sp
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.conditions[?(@.type=="Accepted")].reason`
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// TrafficExtensionPolicy allows the user to configure various traffic extensibility options for the Gateway.
```

Comment:
prefer `EnvoyExtensionPolicy` since these extensions are specific to envoy or `ExtensionPolicy` if these extensions are common across proxy implementations